tests/integration/test_distributeby.py

```diff
@@ -0,0 +1,34 @@
+import dask.dataframe as dd
+import pandas as pd
+import pytest
+
+try:
+    import cudf
+    import dask_cudf
+except ImportError:
+    cudf = None
+    dask_cudf = None
+
+
+@pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])
+def test_distribute_by(c, gpu):
+    df = pd.DataFrame({"id": [0, 1, 2, 1, 2, 3], "val": [0, 1, 2, 1, 2, 3]})
+
+    if gpu:
+        gdf = cudf.from_pandas(df)
+        ddf = dask_cudf.from_cudf(df, npartitions=2)
+    else:
+        ddf = dd.from_pandas(df, npartitions=2)
```

Comment:
Small nitpick but we can probably simplify the dataframe creation a bit:

```python
if gpu:
    xd = cudf
else:
    xd = pd

df = xd.DataFrame({"id": [0, 1, 2, 1, 2, 3], "val": [0, 1, 2, 1, 2, 3]})
ddf = dd.from_pandas(df, npartitions=2)
```

Despite its name, `dd.from_pandas` will create a dask-cudf dataframe if `df` is a cuDF dataframe, which should allow us to circumvent importing dask-cudf here